giab_runner/giab_runner.py

```diff
@@ -0,0 +1,332 @@
+#!/usr/bin/env python3
+
+description = """
+The intent of this script is to make running control samples on specific versions of pipelines easy.
+
+The steps it performs:
+
+1. Check out commit, tag or branch in target repo
+2. Prepare CSV file for the run
+3. Execute the pipeline
+
+It can be configured to run singles, trios and start with FASTQ, BAM and VCF.
+"""
+
+import argparse
+from pathlib import Path
+import subprocess
+import logging
+from pathlib import Path
+import sys
+from logging import Logger
+from configparser import ConfigParser
+from typing import List, Dict, Optional
+
+from help_classes import Case, CsvEntry
+
+
+logging.basicConfig(level=logging.INFO, format="%(levelname)s: %(message)s")
+LOG = logging.getLogger(__name__)
+
+
+def main(
+    config_path: str,
+    label: Optional[str],
+    checkout: str,
+    base_dir: Path,
+    wgs_repo: Path,
+    start_data: str,
+    dry_run: bool,
+    stub_run: bool,
+    run_type: str,
+    skip_confirmation: bool,
+):
+
+    config = ConfigParser()
+    config.read(config_path)
+
+    check_valid_repo(LOG, wgs_repo)
+    check_valid_checkout(LOG, wgs_repo, checkout)
+    checkout_repo(wgs_repo, checkout)
+
+    run_label = build_run_label(run_type, checkout, label, stub_run, start_data)
+
+    results_dir = base_dir / run_label
+    results_dir.mkdir(exist_ok=True, parents=True)
+
+    run_log_path = results_dir / "run.log"
+    write_run_log(run_log_path, run_type, label or "no label", checkout, config)
+
+    if not config.getboolean(run_type, "trio"):
+        csv = get_single_csv(config, run_label, run_type, start_data)
+    else:
+        csv = get_trio_csv(config, run_label, run_type, start_data)
+    out_csv = results_dir / "giab.csv"
+    csv.write_to_file(str(out_csv))
+
+    start_nextflow_command = build_start_nextflow_analysis_cmd(
+        config["settings"]["start_nextflow_analysis"],
+        out_csv,
+        results_dir,
+        stub_run,
+    )
+
+    start_run(start_nextflow_command, dry_run, skip_confirmation)
+
+
+def build_run_label(
+    run_type: str, checkout: str, label: Optional[str], stub_run: bool, start_data: str
+) -> str:
+    label_parts = [run_type]
+    if label is not None:
+        label_parts.append(label)
+    label_parts.append(checkout)
+    if stub_run:
+        label_parts.append("stub")
+    label_parts.append(start_data)
+    run_label = "-".join(label_parts)
+    return run_label
+
+
+def checkout_repo(repo: Path, commit: str):
+
+    LOG.info(f"Checking out: {commit} in {str(repo)}")
+    results = subprocess.run(
+        ["git", "checkout", commit],
+        cwd=str(repo),
+        # text=True is supported from Python 3.7
+        universal_newlines=True,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+    if results.returncode != 0:
+        LOG.error(results.stderr)
+
+
+def check_git_id(repo: Path) -> str:
```

Comment:
```suggestion
def get_git_id(repo: Path) -> str:
```

Yr other `check_` functions do something like exit script if some test fails.

Reply:
Good point 👌